src/components/data/data-utils.js

```diff
@@ -439,14 +441,14 @@ export const EntityUploader = ({ onSuccess, onDismiss, namespace, name, entityTy
             }
           })
         ]),
-        currentFile && _.includes(_.toLower(newEntityType), entityTypes) && div({
+        ((currentFile && entityAlreadyExists) || recordAlreadyExists) && div({
```

Comment:
Is this condition correct? Should it be `currentFile && (entityAlreadyExists || recordAlreadyExists)`? Also, would one state variable `entityOrRecordTypeAlreadyExists` that's used for both providers make some of this logic simpler?

Reply:
`recordAlreadyExists` doesn't depend on whether `currentFile` is true, since the recordType is entered into an input box instead of read from the tsv file.  Also, the following component - a warning about empty cells - doesn't apply to WDS uploads, so it only needs to check the entityType.